src/States.ts

```diff
@@ -2,7 +2,9 @@ export class States <State extends string> implements Iterable<[id: string, stat
   readonly byState: Partial<{ [state in State]: Set<string> }> = {}
 
   set (id: string, state: State): void {
-    this.clear(id, state)
+    if (!this.clear(id, state)) {
```

Comment:
Could the return value be put into a variable that explains what it's supposed to mean?

Right now it reads like "If we didn't clear a previous value, we shouldn't set the new one" and I'm not sure if that's what's supposed to be happening here.

Reply:
Seems reasonable 👍

Reply:
Maybe call the variable 'alreadySet`?